### DISCUSSION TITLE
CSS Flex Masonery layout

### DISCUSSION THREAD

**Discussion by martinlombana** [c-smile/sciter-js-sdk#55] [Q&A] [answered]
How can we achieve, with sciter flex, such a layout? (masonery layout) Whith my example, in CSS3, in the browser we can see this:

![image](https://user-images.githubusercontent.com/18351919/112691159-2c497580-8e7d-11eb-896a-8ac5c6952062.png)

But it is seen like this in sciter-js scapp.exe:

![image](https://user-images.githubusercontent.com/18351919/112691178-38353780-8e7d-11eb-864c-cbffad8d51ac.png)

`
<link rel="stylesheet" type="text/css" href="test.css">

<body>

<div class="container">
  <div class="item" style="height: 140px"></div>
  <div class="item" style="height: 190px"></div>
  <div class="item" style="height: 170px"></div>
  <div class="item" style="height: 120px"></div>
  <div class="item" style="height: 160px"></div>
  <div class="item" style="height: 180px"></div>
  <div class="item" style="height: 140px"></div>
  <div class="item" style="height: 150px"></div>
  <div class="item" style="height: 170px"></div>
  <div class="item" style="height: 170px"></div>
  <div class="item" style="height: 140px"></div>
  <div class="item" style="height: 190px"></div>
  <div class="item" style="height: 170px"></div>
  <div class="item" style="height: 120px"></div>
  <div class="item" style="height: 160px"></div>
  <div class="item" style="height: 180px"></div>
  <div class="item" style="height: 140px"></div>
  <div class="item" style="height: 150px"></div>
  <div class="item" style="height: 170px"></div>
  <div class="item" style="height: 170px"></div>
  
  <span class="item break"></span>
  <span class="item break"></span>
  <span class="item break"></span>
  <span class="item break"></span>
</div>

</body>
`

`.container {
  display: flex;
  flex-flow: column wrap;
  align-content: space-between;
  /* Your container needs a fixed height, and it 
   * needs to be taller than your tallest column. */
  height: 760px;
  
  /* Optional */
  background-color: #f7f7f7;
  border-radius: 3px;
  padding: 20px;
  width: 60%;
  margin: 40px auto;
  counter-reset: items;
}

.item {
  width: 19%;
  /* Optional */
  position: relative;
  margin-bottom: 2%;
  border-radius: 3px;
  background-color: #a1cbfa;
  border: 1px solid #4290e2;
  box-shadow: 0 2px 2px rgba(0,90,250,0.05),
    0 4px 4px rgba(0,90,250,0.05),
    0 8px 8px rgba(0,90,250,0.05),
    0 16px 16px rgba(0,90,250,0.05);
  color: #fff;
  padding: 15px;
  box-sizing: border-box;
}

div.item::before {
  counter-increment: items;
  content: counter(items);
}

/* Re-order items into 3 rows */
.item:nth-of-type(5n+1) { order: 1; }
.item:nth-of-type(5n+2) { order: 2; }
.item:nth-of-type(5n+3) { order: 3; }
.item:nth-of-type(5n+4) { order: 4; }
.item:nth-of-type(5n)   { order: 5; }

/* Force new columns */
.break {
  flex-basis: 100%;
  width: 0;
  border: 1px solid #ddd;
  margin: 0;
  content: "";
  padding: 0;
}

body { font-family: sans-serif; }
h3 { text-align: center; }`

**Comment by c-smile** [comment]
```html
<html>
    <head>
        <title>Test</title>
        <style>

body {
  flow:vertical-wrap;
  overflow-x:scroll-indicator;  
  border-spacing:1em;
}

body > div {
  width:64px;
  background:gold;
}

        </style>
    </head>
    <body>
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:48px" />
    </body>
</html>
```

**Reply by martinlombana** [reply]
Thanks, but this stills gets this result in sciter-js:
![image](https://user-images.githubusercontent.com/18351919/112762801-07cad600-9002-11eb-885d-5e48ace42e4d.png)

Instead of being distributed in 3 Rows via CSS. Is there any equivalent of the above?
![image](https://user-images.githubusercontent.com/18351919/112762811-11543e00-9002-11eb-8802-fc25692b971b.png)


Thanks!

**Comment by c-smile** [comment]
Here is what I see here:

![vertical-wrap](https://sciter.com/wp-content/uploads/2021/03/vertical-wrap.png)

**Reply by martinlombana** [reply]
Nevermind. I see that if the screen is smaller. So your example does work. Nevertheless, my aim was to distribute all the children into 3 columns with css, regardless of the screen height, which is obtained in regular css3 with:

```
/* Re-order items into 3 rows */
.item:nth-of-type(5n+1) { order: 1; }
.item:nth-of-type(5n+2) { order: 2; }
.item:nth-of-type(5n+3) { order: 3; }
.item:nth-of-type(5n+4) { order: 4; }
.item:nth-of-type(5n) { order: 5; }
/* Force new columns */
.break {
flex-basis: 100%;
width: 0;
border: 1px solid #ddd;
margin: 0;
content: "";
padding: 0;
}

```

Your example does work but it is not exactly the distribution that I wanted to achieve. Is there any similar way to distribute all the children into 3 columns?

**Comment by c-smile** [comment, marked as answer]
If you know upfront how columns are split then you can use  `{clear:after | before;}`  to force column wrapping:

```
<html>
    <head>
        <title>Test</title>
        <style>

body {
  flow:vertical-wrap;
  overflow-x:scroll-indicator;  
  border-spacing:1em;
}

body > div {
  width:64px;
  background:gold;
}

body > div:nth-child(4) {clear:after;} 
body > div:nth-child(8) {clear:after;} 

        </style>
    </head>
    <body>
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:48px" />
      <div style="height:64px" />
      <div style="height:48px" />
    </body>
</html>
```

**Comment by martinlombana** [comment]
I see. Thanks for the insight!

___________________________________

*Martín Lombana*

*TODOLOGIC.COM <http://TODOLOGIC.COM> | APPS & MORE*

*+34 633 63 02 35  |   *+34 946 759 137

*INSTAGRAM *→* @ <https://instagram.com/todologic>TODOLOGIC
<https://instagram.com/todologic>*
